pkg/listening-sockets/listening-sockets_test.go

```diff
@@ -0,0 +1,205 @@
+package listeningsockets
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"github.com/openshift-kni/commatrix/pkg/client"
+	"github.com/openshift-kni/commatrix/pkg/consts"
+	"github.com/openshift-kni/commatrix/pkg/types"
+
+	mock_utils "github.com/openshift-kni/commatrix/pkg/utils/mock"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	fakek "k8s.io/client-go/kubernetes/fake"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+type ssTestCase struct {
+	name              string
+	expectedTCPOutput string
+	expectedUDPOutput string
+	expectedssMat     []types.ComDetails
+}
+
+const (
+	tcpExecCommandOutput = (`LISTEN 0      4096      127.0.0.1:8797  0.0.0.0:* users:(("machine-config-",pid=3534,fd=3))                
+	LISTEN 0      4096      127.0.0.1:8798  0.0.0.0:* users:(("machine-config-",pid=3534,fd=13))               
+	LISTEN 0      4096      127.0.0.1:9100  0.0.0.0:* users:(("node_exporter",pid=4147,fd=3))`)
+
+	udpExecCommandOutput = (`UNCONN 0      0           0.0.0.0:111   0.0.0.0:* users:(("rpcbind",pid=1399,fd=5),("systemd",pid=1,fd=78))
+	UNCONN 0      0         127.0.0.1:323   0.0.0.0:* users:(("chronyd",pid=1015,fd=5))                        
+	UNCONN 0      0      10.46.97.104:500   0.0.0.0:* users:(("pluto",pid=2115,fd=21))`)
+
+	procExecCommandOutput = (`1: /system.slice/crio-1234567890abcdef.scope
```

Comment:
Is that something we need to validate?

Reply:
no this is how the output need to be its need to be with crio and a number this is why i created that on that way
this is the output of the cmd command

Reply:
cmd := fmt.Sprintf("cat /proc/%s/cgroup", pid)
output need to have like this
	re := regexp.MustCompile(`crio-([0-9a-fA-F]+)\.scope`)